src/table/virtualized-table/utils/merge-pages.ts

```diff
@@ -1,31 +1,67 @@
-const canMergeRows = (prevPage: EngineAPI.INxPage, page: EngineAPI.INxPage) =>
-  prevPage.qTop + prevPage.qHeight === page.qTop && prevPage.qLeft === page.qLeft && prevPage.qWidth === page.qWidth;
+import { PageInfo } from '../../../types';
+import { COLUMN_DATA_BUFFER_SIZE, ROW_DATA_BUFFER_SIZE } from '../constants';
+import { GridState } from '../types';
 
-const canMergeColumns = (prevPage: EngineAPI.INxPage, page: EngineAPI.INxPage) =>
-  prevPage.qLeft + prevPage.qWidth === page.qLeft && prevPage.qTop === page.qTop && prevPage.qHeight === page.qHeight;
+const isPageStale = (gridState: React.MutableRefObject<GridState>, pageInfo: PageInfo, qPage: EngineAPI.INxPage) => {
+  const pageRowStart = pageInfo.page * pageInfo.rowsPerPage;
+  const colStartIndex = gridState.current.overscanColumnStartIndex - COLUMN_DATA_BUFFER_SIZE;
+  const colEndIndex = gridState.current.overscanColumnStopIndex + COLUMN_DATA_BUFFER_SIZE;
+  const rowStartIndex = pageRowStart + gridState.current.overscanRowStartIndex - ROW_DATA_BUFFER_SIZE;
+  const rowEndIndex = pageRowStart + gridState.current.overscanRowStopIndex + ROW_DATA_BUFFER_SIZE;
+
+  return (
```

Comment:
I think this should be improved. There are really two points here `{ x: qPage.qLeft, y: qPage.qTop }` and `{ x: qPage.qLeft + qPage.qWidth, y: qPage.qTop + qPage.qTop + qPage.qHeight }`. If any of those points are inside the "rectangle" that is the rows and columns that are visible to the user. The page should not be considered stale. This is currently not how this logic work.